### PR TITLE
cxxopts: fixes for conan v2 compatibility in test_package

### PR DESCRIPTION
**Pull request by jcar87**
Specify library name and version:  **cxxopts/all**

Small fixes to support conan2:
* Pass output variable as positional argument to `self.run()`
* Use `self.dependencies` instead of deprecated `self.options`

### CODE REVIEW

**Review by franramirez688 — DISMISSED** [review]

**Review by jwillikers — DISMISSED** [review]

File: recipes/cxxopts/all/test_package/conanfile.py [hunk 1]
```diff
@@ -26,11 +26,11 @@ def test(self):
             output = StringIO()
             bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package")
             option_string = "-f 41 --bar baria --baz"
-            if self.options["cxxopts"].unicode:
+            if self.dependencies["cxxopts"].options.unicode:
```
Comment (SpaceIm) [inline recipes/cxxopts/all/test_package/conanfile.py:29]: it can't work AFAIK

Comment (jcar87) [reply recipes/cxxopts/all/test_package/conanfile.py:29]: indeed! forgot I was in the `test()` method. Will fix!

**Comment by RubenRBS** [comment]
:( v1 seems to fail now and the bot didnt report any v2 runs either

File: recipes/cxxopts/all/test_package/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/cxxopts/all/test_package/conanfile.py:29]: Maybe use ctest trick you've proposed in boost recipe?

Comment (prince-chrismc) [reply recipes/cxxopts/all/test_package/conanfile.py:29]: https://github.com/conan-io/conan-center-index/pull/15091#issuecomment-1372798369 🙏 

This looks super promising from Space!

Comment (prince-chrismc) [reply recipes/cxxopts/all/test_package/conanfile.py]: I am stealing https://github.com/conan-io/conan-center-index/pull/15725 :)

**Review by jwillikers — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]
more compilation problems for older versions can be fixed in the next PR :)

**Pull request merged**